### PR TITLE
Display a list of unavailable apps for updates (aka. merge "Can I update?")

### LINKED ISSUE #2692: Check availablity of apps before doing a server major update
Happened to one of the participants of the Stuttgart meetup.
He updated the instance and suddenly one app was disabled and he couldn't enable it anymore,
because back then the tasks app was not available for NC11.

It should be fairly easy to check this up-front before the person even starts the updater.

What do you think? @LukasReschke @MorrisJobke @jancborchardt

### PR DESCRIPTION
**Pull request by nickvergessen**
> bildschirmfoto von 2018-02-28 14-19-13

Fix #2692

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,6 +34,7 @@
 !/apps/updatenotification
 /apps/updatenotification/js/merged.js
 /apps/updatenotification/js/merged.js.map
+/apps/updatenotification/js/*.hot-update.*
```
Comment (MorrisJobke) [inline .gitignore]: I'm just curious: What is this? 🤔

Comment (nickvergessen) [reply .gitignore]: Don't really know, I just have tones of .hot-update.js and .hot-update.json after running `make build-js` and developing for a while

**Comment by rullzer** [comment]
mmmm how do I test this?

**Comment by nickvergessen** [comment]
Use this:
```diff
diff --git a/apps/updatenotification/lib/Settings/Admin.php b/apps/updatenotification/lib/Settings/Admin.php
index 258cba35e8..9dc08e2a08 100644
--- a/apps/updatenotification/lib/Settings/Admin.php
+++ b/apps/updatenotification/lib/Settings/Admin.php
@@ -86,14 +86,14 @@ class Admin implements ISettings {
                $updateServerURL = $this->config->getSystemValue('updater.server.url', $defaultUpdateServerURL);
 
                $params = [
-                       'isNewVersionAvailable' => !empty($updateState['updateAvailable']),
+                       'isNewVersionAvailable' => true, //FIXME !empty($updateState['updateAvailable']),
                        'isUpdateChecked' => $lastUpdateCheckTimestamp > 0,
                        'lastChecked' => $lastUpdateCheck,
                        'currentChannel' => $currentChannel,
                        'channels' => $channels,
-                       'newVersionString' => empty($updateState['updateVersion']) ? '' : $updateState['updateVersion'],
-                       'downloadLink' => empty($updateState['downloadLink']) ? '' : $updateState['downloadLink'],
-                       'updaterEnabled' => empty($updateState['updaterEnabled']) ? false : $updateState['updaterEnabled'],
+                       'newVersionString' => '14.0.1.3', //FIXME (empty($updateState['updateVersion'])) ? '' : $updateState['updateVersion'],
+                       'downloadLink' => 'https://nextcloud.com/', //FIXME (empty($updateState['downloadLink'])) ? '' : $updateState['downloadLink'],
+                       'updaterEnabled' => true, // FIXME (empty($updateState['updaterEnabled'])) ? false : $updateState['updaterEnabled'],
                        'isDefaultUpdateServerURL' => $updateServerURL === $defaultUpdateServerURL,
                        'updateServerURL' => $updateServerURL,
                        'notifyGroups' => $this->getSelectedGroups($notifyGroups),
```

**Comment by MorrisJobke** [comment]
Some refinements:

* [ ] I needed to check the code to see, that the headings could be clicked to toggle the list below them - for now I would say that they should be just shown and we can look into collapsing them later
* [ ] apps with updates and apps without updates showed the same entries even if the returned API response only listed them in the "missing" section
* [ ] we should show a loading screen, because the info pops up a bit later. I delayed a bit the request and it was  weird, because nothing was shown and after a while it simply popped up - a little spinner with "Checking for app compatibility ..." or something like that would make it a lot better

**Comment by nickvergessen** [comment]
> I needed to check the code to see, that the headings could be clicked to toggle the list below them - for now I would say that they should be just shown and we can look into collapsing them later

missing list is shown by default, I guess that would be okay. Maybe we add a arrow indicating things?

Thanks for the other feedback, will have a look

**Comment by nickvergessen** [comment]
@MorrisJobke resolved all your comments

**Comment by MorrisJobke** [comment]
> @MorrisJobke resolved all your comments

Thanks - will check again :)

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍 

bildschirmfoto 2018-03-06 um 18 33 43

**Comment by MorrisJobke** [comment]
Just a little hint: to build this you need to call `make` in apps/updatenotification.

File: .gitignore (hunk 1 above)
Comment (ChristophWurst) [reply .gitignore]: I guess [hot module replacement](https://webpack.js.org/api/hot-module-replacement/)?

**Review by ChristophWurst — APPROVED** [review]
Code looks good

**Comment by skjnldsv** [comment]
Wording could use a change :)
"2 applications are missing a compatible update"

**Comment by MorrisJobke** [comment]
> Wording could use a change :)
> "2 applications are missing a compatible update"

Still not perfect IMO. Let's get it in for now and maybe refine later :)

**Pull request merged**